File: playground/javascript/node/ingestion.ts
import { ApiError, createIterablePromise } from '@algolia/client-common';
import { ingestionClient } from '@algolia/ingestion';

const appId = process.env.ALGOLIA_APPLICATION_ID || '**** APP_ID *****';
const apiKey = process.env.ALGOLIA_ADMIN_KEY || '**** ADMIN_KEY *****';

// Init client with appId and apiKey
const client = ingestionClient(appId, apiKey, 'us', {
  authMode: 'WithinHeaders',
  hosts: [{ url: 'staging-data.us.algolia.com', accept: 'readWrite', protocol: 'https' }],
});

async function testIngestion() {
  try {
    const res = await client.listAuthentications();

    console.log(`[OK]`, res);
  } catch (e) {
    if (e instanceof ApiError) {
      return console.log(`[${e.status}] ${e.message}`, e.stackTrace, e);
    }

    console.log('[ERROR]', e);
  }
}

testIngestion();

File: playground/ruby/ingestion.rb
require 'dotenv'
require 'algolia'

Dotenv.load('../.env')

client = Algolia::IngestionClient.create(ENV['ALGOLIA_APPLICATION_ID'], ENV['ALGOLIA_ADMIN_KEY'], 'us')

res, status, headers = client.get_authentications_with_http_info(50)
puts res
puts status
puts headers
#res = client.get_tasks()
#puts res.tasks[0]
#res = client.get_task(res.tasks[1].task_id)
#puts res

#res = client.update_task(res.task_id, Algolia::Ingestion::TaskUpdate.new(enabled: false))
#puts res

#res = client.delete_task(res.task_id)
#puts res
